### PR TITLE
Start and Stop Streaming/Recording

### PR DESCRIPTION
**Pull request by ewettach**
Added and tested following functions:

StartStopStreaming - when on execution will start or stop streaming (depending on if the stream status is streaming or not streaming)

StartStopRecording - when on execution will start or stop recording (depending on if the recording status is recording or not recording)

### CODE REVIEW

**Comment by haakonnessjoen** [comment]
Nice :) Maybe you could add presets that uses the start/stop streaming command, and colors the button accordingly with feedback?

**Comment by ewettach** [comment]
I was able to make sure feedback works on recording and streaming as well as added presets for streaming and recording

File: index.js [hunk 1]
```diff
@@ -194,12 +243,24 @@ instance.prototype.action = function(action) {
...
+	switch(action.action){
+		case "set_scene":
+			self.obs.setCurrentScene({
+				'scene-name': action.options.scene
+			});
+		break;
+		case "set_transition":
+			self.obs.setCurrentTransition({
+				'transition-name': action.options.transitions
+			});
```
Comment (haakonnessjoen) [inline index.js]: missing break;

File: index.js [hunk 2]
```diff
@@ -332,7 +472,8 @@ instance.prototype.init_variables = function() {
 	variables.push({ name: 'num_dropped_frames', label: 'Number of dropped frames' });
 	variables.push({ name: 'num_total_frames', label: 'Number of total frames' });
 	variables.push({ name: 'preview_only', label: 'Preview only' });
-	variables.push({ name: 'recording', label: 'Recording state' });
+	// Line commented out due to creating a false state when recording started
+	//variables.push({ name: 'recording', label: 'Recording state' });
```
Comment (haakonnessjoen) [inline index.js]: What does this mean? Doesn't it work?

**Comment by josephdadams** [comment]
Can I help resolve these branch conflicts? I'd really benefit from the new functions that @ewettach has created.

File: index.js (hunk 2 above)
Comment (ewettach) [reply index.js]: I am not sure what is happening here and far from a great programmer, but if if line 

variables.push({ name: 'recording', label: 'Recording state' });

is included it breaks the recording function, and I have not yet found anything useful happening with this line of code included.  Anybody else have any ideas?

**Comment by ewettach** [comment]
@josephdadams I will gladly welcome any help you can offer.  Right now it looks like there is some question on what this line of code does.... see the comments above.

variables.push({ name: 'recording', label: 'Recording state' });

File: index.js (hunk 2 above)
Comment (josephdadams) [reply index.js]: @ewettach I ran your version and uncommented that line and it works fine on my system. Having that variable would be helpful. It starts out as "undefined" though, and doesn't get updated until a command is sent to start or stop recording, and then OBS websockets returns the status. I added a new function when it first connects to request the stream status, which returns booleans for both streaming and recording. If you will give me permissions to add to your fork, I can send that to you.

Comment (josephdadams) [reply index.js]: @ewettach I added a HELP and README to your repo and also did a new PR on your repo to include my changes

**Comment by josephdadams** [comment]
@haakonnessjoen can you review now? I think it’s ready to be merged.

**Comment by josephdadams** [comment]
Bump. :)

**Comment by ewettach** [comment]
@josephdadams @haakonnessjoen Let me know if there is anything I need to do, but I'm assuming it is @haakonnessjoen seat at this point?  Thanks to both of you.

**Comment by josephdadams** [comment]
@haakonnessjoen I added the semicolons back. Ready to merge?

**Comment by josephdadams** [comment]
Thanks!!

**Pull request merged**